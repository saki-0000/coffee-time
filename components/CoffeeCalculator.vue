<template>
  <div>
    <v-text-field
      v-model="water"
      label="お湯の量"
      hint="今日はどのくらい淹れますか？"
      :rules="rules"
      @input="onInputWater()"
    />
    <v-text-field
      v-model="coffee"
      label="コーヒー豆の量"
      hint="今日はどのくらい淹れますか？"
      :rules="rules"
      @input="onInputCoffee()"
    />
    <div class="text">
      コーヒー{{ coffee }}gに必要な お湯の量は{{ water }}mlです。
    </div>
    <div class="text">１湯目：{{ (water * 0.2).toFixed(1) }}ml</div>
    <div class="text">２湯目：{{ (water * 0.2).toFixed(1) }}ml</div>
    <div class="text">３湯目：{{ (water * 0.6).toFixed(1) }}ml</div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api'
import { isNum } from '~/util/helper'
import { useCoffeeCaluculator } from '~/composable/coffee-caluculator'

const { coffee, water, onInputWater, onInputCoffee } = useCoffeeCaluculator()
export default defineComponent({
  setup: () => {
    const validNum = (val: any) => {
      return isNum(val) ? true : '数字で入力してください。'
    }
    const rules = [validNum]
    return { coffee, water, rules, onInputWater, onInputCoffee }
  },
})
</script>
