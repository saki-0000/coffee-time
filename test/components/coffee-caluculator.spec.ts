import { shallowMount, Wrapper } from '@vue/test-utils'
import CoffeeCalculator from '~/components/CoffeeCalculator.vue'

const onInputWater = jest.fn(() => {})
const onInputCoffee = jest.fn(() => {})
jest.mock('~/composable/coffee-caluculator', () => ({
  useCoffeeCaluculator: () => ({
    coffee: 21,
    water: 300,
    onInputWater,
    onInputCoffee,
  }),
}))

describe('CoffeeCalculator.vue', () => {
  let wrapper: Wrapper<any>
  beforeEach(() => {
    wrapper = shallowMount(CoffeeCalculator)
  })

  test('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('call useTime`s functions when the `start` button clicked ', () => {
    wrapper.get('#waterInput').vm.$emit('input')
    expect(wrapper.get('#summuryText').text()).toBe(
      'コーヒー21gに必要な お湯の量は300mlです。'
    )
    wrapper.get('#coffeeInput').vm.$emit('input')
    expect(wrapper.get('#summuryText').text()).toBe(
      'コーヒー21gに必要な お湯の量は300mlです。'
    )
    expect(onInputWater).toHaveBeenCalledTimes(1)
    expect(onInputCoffee).toHaveBeenCalledTimes(1)
  })
})
