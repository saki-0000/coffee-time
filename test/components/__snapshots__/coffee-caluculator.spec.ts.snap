// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoffeeCalculator.vue renders correctly 1`] = `
"<div>
  <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"[Function]\\" successmessages=\\"\\" value=\\"300\\" backgroundcolor=\\"\\" hint=\\"今日はどのくらい淹れますか？\\" id=\\"waterInput\\" label=\\"お湯の量\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" type=\\"text\\"></v-text-field-stub>
  <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"[Function]\\" successmessages=\\"\\" value=\\"21\\" backgroundcolor=\\"\\" hint=\\"今日はどのくらい淹れますか？\\" id=\\"coffeeInput\\" label=\\"コーヒー豆の量\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" type=\\"text\\"></v-text-field-stub>
  <div id=\\"summuryText\\" class=\\"text mb-3\\">
    コーヒー21gに必要な お湯の量は300mlです。
  </div>
  <div id=\\"firstText\\" class=\\"text\\">
    １湯目：60.0ml
  </div>
  <div id=\\"secondText\\" class=\\"text\\">
    ２湯目：60.0ml
  </div>
  <div id=\\"thirdText\\" class=\\"text\\">
    ３湯目：180.0ml
  </div>
</div>"
`;
